### PR TITLE
New functions upper() and lower() with two parameters (text, language)

### PR DESCRIPTION
**Pull request by bbarany**
Hi,

I added additional functions for the upper(text, language) and lower(text, language) syntax of DB2.
The second argument is ignored, as Unicode doesn't really have a concept of language-specific upper or lower case.

I hope I got the version update right, please double-check.

Regards,
Balázs

### CODE REVIEW

File: db2fce--0.0.11--0.0.12.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+-- UPPER(text, text): DB2 allows the user to specify a language for the UPPER function.
+-- This is not really needed with PostgreSQL, as upper() uses Unicode, independent of the language.
+-- On DB2, upper('Müller', 'en_US') also returns MÜLLER, just as with de_DE.
+CREATE FUNCTION db2.upper(TEXT, TEXT) RETURNS TEXT AS
```
Comment (mbanck) [inline db2fce--0.0.11--0.0.12.sql:4]: This change is in both  db2fce--0.0.11--0.0.12.sql and  db2fce--0.0.12.sql → db2fce--0.0.13.sql  and I think it should only be in the latter - if you'd update from 0.0.11 to 0.0.13, you'd run the ```CREATE FUNCTION db2.upper()``` twice resulting in a error, unless I am missing something?

Uh, I see what's going on now - I misread the "rename db2fce--0.0.12.sql to db2fce--0.0.13.sql"  as the corresponding update script.

Still, as 0.0.12 is not yet released, the above rename should not happen - I'd prefer a patch which changes 0.0.12 and optionally (only if you insist, I think it'd be cleaner to let the maintainer do this) prepares a release, but we'd need to tag/release first before we rename things to 0.0.13 in any case.

**Comment by mbanck** [comment]
I commented on the last commit a bit, but to summarize here:

I'd be happy to cherry-pick a commit which implements the UPPER() and LOWER() functions (including the test suite changes) and would then do a 0.0.12 release. The current branch includes a lot of outdated commits which are already cherry-picked or mostly identically in master, so I won't merge it as is.

Alternatively, I can just adopt a commit for UPPER() and LOWER() myself and credit it to you and then do a release, if that'd be easier.

File: expected/string_text.out [hunk 2]
```diff
@@ -49,3 +49,17 @@ ERROR:  function hex(integer) does not exist
...
+--------
+ MÜLLER
+(1 row)
+
+-- LOWER(text, text)
+SELECT LOWER('MÜLLER', 'de_DE') FROM SYSIBM.SYSDUMMY1;
+ lower  
+--------
+ müller
+(1 Zeile)
```
Comment (mbanck) [inline expected/string_text.out:65]: Wrong language (```1 Zeile``` vs.  ```1 row```)

**Comment by mbanck** [comment]
I've adopted your changes as b0910e189380646dbd64e311e8abdb1392d609de and released v0.0.12

**Comment by bbarany** [comment]
Thank you for the fixes and the release. Your way of organizing the patch is better.

**Pull request closed without merging**